Multisig/UI/App/UIKit/AppSettingsViewController/AppSettingsViewController.swift

```diff
@@ -0,0 +1,226 @@
+        tableView.registerHeaderFooterView(BasicHeaderView.self)
+
+        buildSections()
+
+        notificationCenter.addObserver(
+            self, selector: #selector(hidePresentedController), name: .ownerKeyImported, object: nil)
+    }
+
+    private func buildSections() {
+        let signingKey = AppSettings.current().signingKeyAddress
+        sections = [
+            (section: .general, items: [
+                signingKey != nil ?
+                    Section.General.importedKey("Imported owner key", signingKey!) :
+                    Section.General.importKey("Import owner key"),
+                Section.General.terms("Terms of use"),
+                Section.General.privacyPolicy("Privacy policy"),
+                Section.General.licenses("Licenses"),
+                Section.General.getInTouch("Get in touch"),
+                Section.General.appVersion("App version", app.marketingVersion),
+                Section.General.network("Network", app.network.rawValue),
+            ]),
+            (section: .advanced, items: [Section.Advanced.advanced("Advanced")])
+        ]
+    }
+
+    @objc func hidePresentedController() {
+        presentedViewController?.dismiss(animated: true)
+        reload()
+    }
+
+    private func reload() {
+        buildSections()
+        tableView.reloadData()
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        sections.count
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        sections[section].items.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let item = sections[indexPath.section].items[indexPath.row]
+        switch item {
+        case Section.General.importKey(let name):
+            return basicCell(name: name, indexPath: indexPath)
+        case Section.General.importedKey(let name, let signingKey):
+            return importedKeyCell(name: name, signingKey: signingKey, indexPath: indexPath)
+        case Section.General.terms(let name):
+            return basicCell(name: name, indexPath: indexPath)
+        case Section.General.privacyPolicy(let name):
+            return basicCell(name: name, indexPath: indexPath)
+        case Section.General.licenses(let name):
+            return basicCell(name: name, indexPath: indexPath)
+        case Section.General.getInTouch(let name):
+            return basicCell(name: name, indexPath: indexPath)
+        case Section.General.appVersion(let name, let version):
+            return infoCell(name: name, info: version, indexPath: indexPath)
+        case Section.General.network(let name, let network):
+            return infoCell(name: name, info: network, indexPath: indexPath)
+        case Section.Advanced.advanced(let name):
+            return basicCell(name: name, indexPath: indexPath)
```

Comment:
this is really unreadable blob of case statements. I think that if there's a blank line before each case statement, it will be better readable

Reply:
ok